Redirect Chain Lightning to Grounding Totem as well

kartoffelsup opened the pull request.

killerwife commented:
Interesting

kartoffelsup commented:
Not sure if this has any negative impact on other chained damage spells, since I do not know of any other than Chain Lightning. If you can tell me how to get a list of all chained damage spells, I will thoroughly test those as well.

robinsch commented:
This will break all chain spells.

kartoffelsup commented:
How will it break all of them?

robinsch commented:
Because it will always search of chain targets if magnet returns nullptr, even if EffectChainTarget <= 1

kartoffelsup commented:
So what you're saying is, I should chain those two conditions with an or? :< Or would that not solve it?

robinsch commented:
No, it should simply just set the magnet target if we found one. There is no else condition.

kartoffelsup commented:
It appears that I misunderstood the magnet target.

kartoffelsup commented:
I'm sorry, I'll try to find a better solution :/

killerwife commented:
Also, should you wanna dynamically search spell dbc, i suggest you load it up into SQL. Works like a charm.

robinsch commented:
I prefer using QSW (https://github.com/anathemareborn/QSpellWork/releases) or download the source if you want to change something.

killerwife commented:
Switch case might be a way to go until we find out a more general behaviour. And QSW is fine, until you are doing SpellAttribute research and you need to be quick about it.

kartoffelsup commented:
Does this look better now?
>you need to change other calls where the target is set by this function
I let VisualStudio find all references (I hope this is reliable) and the only other place where it wasn't used with 
`if (Unit* pUnitTarget = m_caster->SelectMagnetTarget(target, this, effIndex))`
was Unit.cpp#2261 wich I adjusted within my previous commit.

robinsch commented:
All you have to do is set the target to magnet if we found a magnet, otherwise the target is stored in m_targets.getUnitTarget() and execute code https://github.com/cmangos/mangos-tbc/blob/master/src/game/Spell.cpp#L1670 if EffectChainTarget > 1

kartoffelsup commented:
Not sure if I understand you correctly, but it sounds like I did exactly what you describe here. I just removed the giant if/else block and replaced it with if/breaks.

robinsch commented:
Then it should be fine, but please check the second paragraph. I'm pretty sure this breaks all spells with TARGET_SINGLE_ENEMY and TARGET_DUELVSPLAYER

kartoffelsup commented:
Alright, I will check all calls to select magnet target again to make sure! :)

kartoffelsup commented:
It this what you had in mind?

robinsch commented:
Yes, this should be fine. (Except for coding style)

kartoffelsup commented:
You mean the curly braces for if/else with only a single statement?

robinsch commented:
Yes

kartoffelsup commented:
Darn, I always forget them :( 
I will apply the code style asap.

robinsch commented on src/game/Spell.cpp:
```diff
@@ -1659,70 +1659,76 @@ void Spell::SetTargetMap(SpellEffectIndex effIndex, uint32 targetMode, UnitList&
 
-                tempTargetUnitMap.sort(TargetDistanceOrderNear(pUnitTarget));
+            UnitList tempTargetUnitMap;
+            {
```
?

kartoffelsup commented:
How should I proceed here?
